src/common.cc

```diff
@@ -126,7 +126,7 @@ stim::DetectorErrorModel common::remove_zero_probability_errors(
         out_dem.append_dem_instruction(instruction);
         break;
       default:
-        assert(false && "unreachable");
+        std::cout << "Unrecognized instruction type: " << instruction.type << std::endl;
```

Comment:
Once all the known `stim::DemInstructionType` are handled, the default could throw an exception (which pybind will convert into a ValueError).

```suggestion
        throw std::invalid_argument("Unrecognized instruction type: " + op.str());
```

That means there should be a separate case for each of these:
```
enum class DemInstructionType : uint8_t {
    DEM_ERROR,
    DEM_SHIFT_DETECTORS,
    DEM_DETECTOR,
    DEM_LOGICAL_OBSERVABLE,
    DEM_REPEAT_BLOCK,
};
```

of which all but `DEM_ERROR` and `DEM_DETECTOR` can be silently ignored (no need for `std::cout`).